## The difference between t and z tests is that with a t test, the population standard deviation is not known!
### We therefore approximate the population standard deviation from our sample. Accounting for this change leads to an additional paremeter used in the test, namely, the "degrees of freedom" which influences the rejection zones. The rest of the test is the same.

#### Example: machine bottles 150cc, 4 bottles randomly picked and average volume found to be 151cc and a sample sdt dev of 2cc. Has the mean volume changed?

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats
from statsmodels.stats import weightstats

In [3]:
volumes = pd.Series([148.5, 153.4, 150.9, 151.2])
volumes.describe()

count      4.000000
mean     151.000000
std        2.004994
min      148.500000
25%      150.300000
50%      151.050000
75%      151.750000
max      153.400000
dtype: float64

In [6]:
stats.ttest_1samp(volumes, 150) # degrees of freedom here is 3, since (4-1) samples 

Ttest_1sampResult(statistic=0.997509336107632, pvalue=0.3920333832606524)

### the first result is the test statistic ($t_{calc} = \frac{\bar{x}-\mu}{\frac{s}{\sqrt{n}}}$), the second result is the p value
### since p = 0.392 > 0.05 we fail to reject the null hypothesis. From this test we cannot conclude that the mean value of 150 per bottle has changed